framework/resource/manager.go

```diff
@@ -363,6 +363,9 @@ func WithResource(r *Resource) ManagerOption {
 			name, _ = component.Id()
 		}
 
+		// Note(izaak): If multiple resources have the same name, all but one
+		// will be overwritten. We could enforce uniqueness here, but we'd have
+		// to introduce an error return.
```

Comment:
Hmm, if we're expecting uniqueness then maybe we should update the function to return an error? I could see this causing some fun bugs if a platform has resources with the same name (but some other unique identifier)

Reply:
This isn't new behavior that i'm introducing - we had this problem before this change. I could see opening a separate issue to fix this, in addition to this comment.

Reply:
Yeah! I'd open an issue 👍🏻